showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/logging/ITLogging.java

```diff
@@ -0,0 +1,207 @@
+import com.google.showcase.v1beta1.EchoRequest;
+import com.google.showcase.v1beta1.EchoResponse;
+import com.google.showcase.v1beta1.it.util.TestClientInitializer;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.event.KeyValuePair;
+
+// This test needs to run with GOOGLE_SDK_JAVA_LOGGING=true
+//  mvn clean verify -P '!showcase,enable-integration-tests,loggingTestBase,slf4j2_logback'
+public class ITLogging {
+  private static EchoClient grpcClient;
+
+  private static EchoClient httpjsonClient;
+
+  private static final KeyValuePair SERVICE_NAME_KEY_VALUE_PAIR =
+      new KeyValuePair("serviceName", "google.showcase.v1beta1.Echo");
+  private static final KeyValuePair RPC_NAME_KEY_VALUE_PAIR =
+      new KeyValuePair("rpcName", "google.showcase.v1beta1.Echo/Echo");
+  private static final KeyValuePair RESPONSE_STATUS_KEY_VALUE_PAIR =
+      new KeyValuePair("response.status", "OK");
+  private static final KeyValuePair RESPONSE_STATUS_KEY_VALUE_PAIR_HTTP =
+      new KeyValuePair("response.status", "200");
+
+  private static final KeyValuePair RESPONSE_HEADERS_KEY_VALUE_PAIR =
+      new KeyValuePair("response.headers", ImmutableMap.of("content-type", "application/grpc"));
+
+  private static final String ECHO_STRING = "echo?";
+
+  private static Logger logger = LoggerFactory.getLogger(ITLogging.class);
+
+  private TestAppender setupTestLogger(Class<?> clazz) {
+    TestAppender testAppender = new TestAppender();
+    testAppender.start();
+    ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+    listAppender.start();
+    org.slf4j.Logger logger = LoggerFactory.getLogger(clazz);
+    ((ch.qos.logback.classic.Logger) logger).setLevel(Level.DEBUG);
+    ((ch.qos.logback.classic.Logger) logger).addAppender(testAppender);
+    return testAppender;
+  }
+
+  @BeforeAll
+  static void createClients() throws Exception {
+    // Create gRPC Echo Client
+    grpcClient = TestClientInitializer.createGrpcEchoClient();
+    // Create Http JSON Echo Client
+    httpjsonClient = TestClientInitializer.createHttpJsonEchoClient();
+  }
+
+  @AfterAll
+  static void destroyClients() throws InterruptedException {
+    grpcClient.close();
+    httpjsonClient.close();
+
+    grpcClient.awaitTermination(TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+    httpjsonClient.awaitTermination(
+        TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  @Test
+  void testGrpc_receiveContent_logDebug() {
+    TestAppender testAppender = setupTestLogger(GrpcLoggingInterceptor.class);
+    assertThat(echoGrpc(ECHO_STRING)).isEqualTo(ECHO_STRING);
+
+    assertThat(true).isTrue();
+
+    assertThat(testAppender.events.size()).isEqualTo(2);
```

Comment:
Can you leave a comment explaining a few of these magic numbers. Why is events.size() == 2 here and events.size() == 4 below?

Reply:
events.size()== 2 --> one for request and one for respond, each tested in below lines.
there is no events.size() == 4, I suppose you meant `assertThat(keyValuePairs.size()).isEqualTo(4);`?
similarly, values from keyValuePairs are checked in more details below that.